community/planning/grid_document.md

```diff
@@ -0,0 +1,808 @@
+constraints. Grid Document transactions may appear as follows:
+
+##### CreateFolderAction
+The action to create a folder will include a `workflow_id` and `workflow_state`
+field. These fields must refer to an existing workflow and workflow state for the
+transaction to be valid.  Additionally, the agent submitting the transaction must
+have the workflow permission `document::can-create-folder` within the specified
+workflow state.
+
+```
+message CreateFolderAction {
+  …
+  string workflow_id = 4;
+  string workflow_state = 5;
+}
+```
+
+##### DeleteFolderAction
+The transaction payload to delete a folder does not need to be extended to
+support workflow permissions. An agent attempting to delete a folder must
+have the `document::can-delete-folder` permission for the current workflow
+state of the folder. If the agent does not have this permission within the
+folder’s workflow state, the transaction is invalid.
+
+##### CreateFileAction
+The transaction to create a file will be extended to include a `workflow_state`.
+This allows Grid Document to verify the submitting agent has the
+`document::can-create-file` within the indicated `workflow_state`. Furthermore,
+Grid Document will validate that the submitting agent has the
+`document::can-access-folder` permission for the folder in its current workflow
+state. If the agent does not have either of these permissions, the transaction
+is invalid. The transaction payload may appear as follows:
+
+```
+message CreateFileAction {
+  …
+  string workflow_state = 4;
+}
+```
+
+##### DeleteFileAction
+The transaction payload to delete a file will also remain unchanged to support
+workflow permissions. An agent attempting to delete a file must have the
+`document::can-delete-file` permission within the file’s current workflow state.
+Additionally, Grid Document will validate that the agent has the
+`document::can-access-folder` permission for the parent folder’s current workflow
+state. If the submitting agent does not have either of these permissions, the
+transaction is invalid.
+
+## Drawbacks
+[drawbacks]: #drawbacks
+
+This change will increase the size of state for grid instances. There is not a
+way around this as the files have to be saved somewhere. This necessitates the
+ability to groom state, i.e., deleting files and folders. However, based on
+Grid’s storage backend, this may cause issues with large amounts of historical
+data being persisted.
+
+## Rationale and alternatives
+[alternatives]: #alternatives
+
+### Rationale
+
+* Grid Document allows for secure document sharing
+
+* Light-weight implementation
+
+* Very familiar and simple use-case to ease users in to Grid
+
+### Alternatives
+
+* Users may use e-mail or online document sharing sites. This does not
+  necessarily provide the same amount of security as Grid Document.
+
+* Enforce permissioning, as seen in other smart contracts. This proves to be a
```

Comment:
I'm not sure this is in the right section - i.e. this is not an alternative to Grid Doc.